### PR TITLE
plugin iam role has tags complete

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -327,6 +327,7 @@ module.exports = {
         'rolePolicyUnusedServices'      : require(__dirname + '/plugins/aws/iam/rolePolicyUnusedServices.js'),
         'iamUserInUse'                  : require(__dirname + '/plugins/aws/iam/iamUserInUse.js'),
         'policyAllowsToChangePassword'  : require(__dirname + '/plugins/aws/iam/policyAllowsToChangePassword.js'),
+        'iamRolesHasTags'              : require(__dirname + '/plugins/aws/iam/iamRolesHasTags.js'),
```
Comment (AkhtarAmir) [inline exports.js:330]:
```suggestion
        'iamRoleHasTags'               : require(__dirname + '/plugins/aws/iam/iamRoleHasTags.js'),
```

File: plugins/aws/iam/iamRolesHasTags.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+var helpers = require('../../../helpers/aws');
+
```
Comment (AkhtarAmir) [inline plugins/aws/iam/iamRolesHasTags.js:2]: You will have to change file name as well

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: exports.js [hunk 3]
```diff
@@ -327,6 +327,7 @@ module.exports = {
         'rolePolicyUnusedServices'      : require(__dirname + '/plugins/aws/iam/rolePolicyUnusedServices.js'),
         'iamUserInUse'                  : require(__dirname + '/plugins/aws/iam/iamUserInUse.js'),
         'policyAllowsToChangePassword'  : require(__dirname + '/plugins/aws/iam/policyAllowsToChangePassword.js'),
+        'iamRoleHasTags'              : require(__dirname + '/plugins/aws/iam/iamRoleHasTags.js'),
```
Comment (AkhtarAmir) [inline exports.js:330]:
```suggestion
        'iamRoleHasTags'                : require(__dirname + '/plugins/aws/iam/iamRoleHasTags.js'),
```

File: exports.js [hunk 4]
```diff
@@ -820,6 +821,7 @@ module.exports = {
         'automaticFailoverEnabled'      : require(__dirname + '/plugins/azure/cosmosdb/automaticFailoverEnabled.js'),
 
         'checkAdvisorRecommendations'   : require(__dirname + '/plugins/azure/advisor/checkAdvisorRecommendations.js'),
+
```
Comment (AkhtarAmir) [inline exports.js:824]:
```suggestion
```

File: plugins/aws/iam/iamRoleHasTags.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    title: 'IAM Roles Have Tags',
+    category: 'IAM',
+    domain: 'Identity and Access management',
+    description: 'Ensure that AWS IAM Roles have tags associated.',
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    link: 'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_tags.html',
+    recommended_action: 'Modify Roles to add tags.',
+    apis: ['IAM:listRoles'],
+    run: function(cache, settings, callback) {
+
```
Comment (alphadev4) [inline plugins/aws/iam/iamRoleHasTags.js:13]:
```suggestion
```

File: plugins/aws/iam/iamRoleHasTags.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        var results = [];
+        var source = {};
+
```
Comment (alphadev4) [inline plugins/aws/iam/iamRoleHasTags.js:16]:
```suggestion
```

File: plugins/aws/iam/iamRoleHasTags.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            helpers.addResult(results, 0, 'No IAM roles found');
+            return callback(null, results, source);
+        }
+        for (var role of listRoles.data){
+            const {Arn, Tags} = role;
+            if (!Tags.length) {
+                helpers.addResult(results, 2, 'IAM Role does not have tags', Arn);
+            } else {
+                helpers.addResult(results, 0, 'IAM Role have tags', Arn);
+
```
Comment (alphadev4) [inline plugins/aws/iam/iamRoleHasTags.js:38]:
```suggestion
```

File: plugins/aws/iam/iamRoleHasTags.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+            helpers.addResult(results, 3,
+                'Unable to query for IAM roles: ' + helpers.addError(listRoles));
+            return callback(null, results, source);
+        }
+
+        if (!listRoles.data.length) {
+            helpers.addResult(results, 0, 'No IAM roles found');
+            return callback(null, results, source);
+        }
+        for (var role of listRoles.data){
```
Comment (alphadev4) [inline plugins/aws/iam/iamRoleHasTags.js:32]:
```suggestion
        for (var role of listRoles.data) {
```

**Review by alphadev4 — APPROVED** [review]

File: plugins/aws/iam/iamRoleHasTags.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+            return callback(null, results, source);
+        }
+
+        if (!listRoles.data.length) {
+            helpers.addResult(results, 0, 'No IAM roles found');
+            return callback(null, results, source);
+        }
+        for (var role of listRoles.data) {
+            const {Arn, Tags} = role;
+            if (!Tags.length) {
```
Comment (AkhtarAmir) [inline plugins/aws/iam/iamRoleHasTags.js:34]:
```suggestion
            if (!Tags || !Tags.length) {
```

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**